docs/content/en/reference/native-extensions.md

```diff
@@ -40,6 +40,9 @@ Once a connection has been established stencil communicates with the plugin over
 3. When the function is called via `extensions.Call "importPath.function"` the rpc `ExecuteTemplateFunction` is called with the arguments passed to the function. The format for this RPC is defined as [`TemplateFunctionExec`](https://pkg.go.dev/github.com/getoutreach/stencil/pkg/extensions/apiv1#TemplateFunctionExec) in Go.
 4. The response from the RPC is returned directly to the go-template with no processing.
 
+## Debug Native Extension
+The [go-plugin](https://github.com/hashicorp/go-plugin) library does not surface error to stencil. You could just get general error message of `failed to create connection to extension: Unrecognized remote plugin message`. In this case, run the native extension binary directly to find out the error in the plugin.
+
```

Comment:
suggestion: this should go directly below "Testing a Native Extension" and the heading should match the same verb tense as the other headings.
```suggestion
## Debugging a Native Extension
The [`go-plugin`](https://github.com/hashicorp/go-plugin) library does not surface errors to stencil. Instead, it will raise the generic message `failed to create connection to extension: Unrecognized remote plugin message`. To determine the correct error message for the plugin, execute the native extension binary directly.

```